### DISCUSSION TITLE
An issue with Using CSLA 5: Blazor and Web Assembly Book

### DISCUSSION THREAD

**Discussion by BuckHicks** [MarimerLLC/csla#2137] [Questions] [answered]
I just started a job that uses CSLA for everything so I have been reading everything I can about it and working through any tutorials that I can find. Although most of the material I have is outdated I stumbled across the "Using CSLA 5: Blazor and Web Assembly Book", which was a spectacular find because  I am new to both platforms.

So anyway I made it to page 93 before I ran into a problem that I can't resolve. At the top of the page under the **MainLayout Content** heading it says to add @inject CurrentUserService CurrentUserService to the page and 

`protected override void OnInitialized()
{
CurrentUserService.CurrentUserChanged +=
(sender, args) => StateHasChanged();
base.OnInitialized();
}`

But I can't figure what page is being referenced here. I have looked at the MainLayout page and some of the other pages on Git and do not see either of these there. Is this an error in the text or am I missing something from it?

Has anybody else worked through this book and ran into this issue?

**Comment by rockfordlhotka** [comment, marked as answer]
The [MainLayout page](https://github.com/MarimerLLC/BlazorBook/blob/master/BlazorAuthClient/BlazorAuthClient/Client/Shared/MainLayout.razor) is in the /Shared directory. I believe that part of the chapter is an intermediate step on the way to the final result.

In the end, it is the [Login page that uses the CurrentUserService](https://github.com/MarimerLLC/BlazorBook/blob/master/BlazorAuthClient/BlazorAuthClient/Client/Pages/Login.razor).

**Reply by BuckHicks** [reply]
Thank you. I was just confused by the heading but you helped me solve the problem. If anyone else gets confused by this. The  @Inject CurrentUserService CurrentUserService goes in the login page and the OnInitialized block goes in the logout page not the MainLayout file.
